optout/src/test/java/gov/cms/ab2d/optout/OptOutProcessorIntegrationTest.java

```diff
@@ -0,0 +1,66 @@
+package gov.cms.ab2d.optout;
+
+import gov.cms.ab2d.common.model.Consent;
+import gov.cms.ab2d.common.repository.ConsentRepository;
+import gov.cms.ab2d.optout.gateway.S3Gateway;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.Matchers.empty;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@Slf4j
+@SpringBootTest
+class OptOutProcessorIntegrationTest {
+
+    @MockBean
+    private S3Gateway mockS3Gateway;
+
+    @Autowired
+    private ConsentRepository consentRepo;
+
+    @Autowired
+    private ConsentConverterService consentConverterSvc;
+
+    private OptOutProcessor cut;
+
+
+    @BeforeEach
+    void setUp() {
+        cut = new OptOutProcessorImpl(mockS3Gateway, consentRepo, consentConverterSvc);
```

Comment:
I don't think we need to wire the `OptOutProcessorImpl` manually here. Just declare it among the `@Autowired` ones above. Spring will recognize that you want to use S3 mock bean and _will_ inject that for you.

Reply:
done